lib.typ

```diff
@@ -225,10 +226,40 @@
     })
   }
 
+  // Transforms the following:
+  // ```md
+  // ---
+  // frontmatter
+  // ---
+  // content
+  // ```
+  // into the tuple (frontmatter, content) of type (str, str).
+  //
+  // The frontmatter is left as a string to let the user
+  // decide the metadata format i.e. yaml, toml, etc.
```

Comment:
We should add a test for this. Create the file `tests/yaml.typ`; you should be able to see the other `tests/*.typ` files for how to structure it (run `cargo test --workspace` to run the tests).